packages/cozy-client/src/CozyClient.js

```diff
@@ -1106,7 +1106,20 @@ client.query(Q('io.cozy.bills'))`)
    * @param {CozyClientDocument | Array<CozyClientDocument>} data - Document or array of documents to be saved
    * @returns {Promise<void>}
    */
-  async persistVirtualDocuments(data) {
+  async persistVirtualDocuments(definition, data) {
+    if (definition.doctype === 'io.cozy.apps_registry') {
+      // io.cozy.apps_registry has a dedicated endpoint on cozy-stack that
+      // returns data different than the one stored in database
+      // We want to store the full answer data under the `maintenance` id
+      // so we can query it from the Pouch the same way we query it from the stack
+      return await this.persistVirtualDocument({
```

Comment:
I had some trouble to understand the behavior at first read. I would propose a slight modification in the comment,, to make it clear that the query returns an array of documents: 
```js
      // io.cozy.apps_registry has a dedicated `maintenance` endpoint on cozy-stack
      //  that returns data differently than the one stored in database.
      // As we want to have transparent queries, whether it uses the stack API or Pouch,
      // we store the full response into a single doc, with a `maintenance`  _id 
      // and a special `cozyPouchData` attribute, to highlight this special case.
   ```

Reply:
Fixed here: https://github.com/cozy/cozy-client/compare/d5677b53a2195955451f35bf97a3e32d4c0990e8..95978d39546023920b0c01d689fed5dd41577a02#diff-9d26647e8d41ba1414842f5d3dca72250075f14efed52997e19584197a66e2a6